scripts/install.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+
+CHAIN_ID=''
+MONIKER=''
+PROVENANCED_VERSION=''
+
+SYSTEMD_FILE='[Unit]
+Description=provenanced
+Requires=network-online.target
+After=network-online.target
+
+[Service]
+Restart=on-failure
+User=provenanced
+Group=provenanced
+PermissionsStartOnly=true
+ExecStart=/home/provenanced/go/bin/cosmovisor start -t --home /provenanced/
+Environment="PATH=/home/provenanced/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/go/bin"
+Environment="DAEMON_NAME=provenanced"
+Environment="DAEMON_HOME=/provenanced"
+Environment="DAEMON_RESTART_AFTER_UPGRADE=true"
+Environment="DAEMON_ALLOW_DOWNLOAD_BINARIES=true"
+ExecReload=/bin/kill -HUP $MAINPID
+KillSignal=SIGTERM
+
+[Install]
+WantedBy=multi-user.target'
+
+function check_err () {
+  if [[ $1 -ne 0 ]]; then
+    echo -e "\033[0;31m ✕\033[0m"
+    echo -e "\033[0;31m$2\033[0m"
+    exit 1
+  else
+    echo -e "\033[0;32m ✓\033[0m"
+  fi
+}
+
+function append_ifnoe () {
+  su $3 -c "grep -qxF \"$1\" $2" || echo \"$1\" | su $3 -c "tee -a $2"
+}
+
+printf "Checking current user... "
+if [[ "${EUID}" -ne 0 ]]; then
+  check_err 1 "user should be root"
+else
+  check_err 0 ""
+fi 
+
+printf "Installing pre-requisite libraries... "
+err=$(apt-get update && apt-get install -y jq unzip libuser libleveldb-dev git curl systemd)
+check_err $? "${err}"
+
+printf "Deleting existing go install... "
+err=$(rm -rf /usr/local/go 2>&1 > /dev/null)
+check_err $? "${err}"
+
+printf "Downloading go... "
+err=$(curl -L -s https://golang.org/dl/go1.15.8.linux-amd64.tar.gz -o /tmp/go1.15.8.linux-amd64.tar.gz 2>&1 > /dev/null)
+check_err $? "${err}"
+
+printf "Unpacking go... "
+err=$(tar -xf /tmp/go1.15.8.linux-amd64.tar.gz -C /usr/local/ 2>&1 > /dev/null)
+check_err $? "${err}"
+
+printf "Creating provenanced group... "
+err=$(id -g provenanced 2>&1 > /dev/null || lgroupadd provenanced)
+check_err $? "${err}"
+
+printf "Creating provenanced service user... "
+err=$(id -u provenanced 2>&1 > /dev/null || luseradd provenanced -g provenanced -d /home/provenanced -s /bin/bash)
+check_err $? "${err}"
+
+printf "Updating provenanced user profile... "
+for line in 'GOPATH="$HOME/go"' 'PATH="$GOPATH/bin:$PATH"' 'PIO_HOME="/provenanced/"'; do
+  err=$(append_ifnoe ${line} /home/provenanced/.profile provenanced 2>&1 > /dev/null)
+  if [[ $? -ne 0 ]]; then
+    check_err 1 "${err}"
+  fi
+done
+check_err 0 ""
+
+printf "Downloading cosmovisor... "
+err=$(su provenanced -c "/usr/local/go/bin/go get github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor 2>&1 > /dev/null")
```

Comment:
So the only thing here is we are now using a forked version of cosmovisor that Phil created. github.com/provenance-io/cosmovisor/cmd/cosmovisor"